src/backend/metal/src/device.rs

```diff
@@ -1195,8 +1207,8 @@ impl hal::device::Device<Backend> for Device {
                                     .contains(n::DescriptorContent::IMMUTABLE_SAMPLER)
                                 {
                                     let immutable_sampler = &immutable_samplers[&layout.binding];
-                                    let handle =
-                                        inline_samplers.append(immutable_sampler.data.clone());
+                                    let handle = inline_samplers.len() as u8;
```

Comment:
nit: here and in the push constants, we should ideally use the types, like `naga::back::msl::InlineSamplerIndex` and `naga::back::msl::Slot` instead of hard-coding `u8`.
This is super minor and non-blocking.